libsolidity/analysis/ControlFlowBuilder.cpp

```diff
@@ -300,8 +300,7 @@ bool ControlFlowBuilder::visit(FunctionCall const& _functionCall)
 				_functionCall.expression().accept(*this);
 				ASTNode::listAccept(_functionCall.arguments(), *this);
 
-				solAssert(!m_currentNode->functionCall);
-				m_currentNode->functionCall = &_functionCall;
+				m_currentNode->functionDefinition = ASTNode::resolveFunctionCall(_functionCall, m_contract);
```

Comment:
Can you quickly explain why `m_contract` is always the same as `item.first.contract` and `item.contract` below? In other words, is this always done for a fixed "final contract"?

Reply:
When we build the CFG for a function we give the builder a contract and the builder puts it in `m_contract`. Then we put that CFG under (contract, function) in `CFG::m_functionControlFlow`. The contract in the pair is always the same thing we passed in:
https://github.com/ethereum/solidity/blob/2f33105d11ae4f5b76a352abc92a331ee2a1e0b7/libsolidity/analysis/ControlFlowGraph.cpp#L46-L47

In the two places you're asking about, we use `CFG::functionFlow()` to pull the CFG corresponding to (`item.contract`, `item.function`) from `CFG::m_functionControlFlow`:
https://github.com/ethereum/solidity/blob/2f33105d11ae4f5b76a352abc92a331ee2a1e0b7/libsolidity/analysis/ControlFlowGraph.cpp#L52-L55

If the CFG is stored under `item.contract` then that must have been the value of `m_contract` when the CFG was being built.

> In other words, is this always done for a fixed "final contract"?

No, we have a separate CFGs for each contract and we refer to them with a full (contract, function) pair. So here you'll get different CFGs for different values of `item.contract` and each one will have the function resolved correctly from the perspective of that contract.

I.e. the contract is only fixed for a specific `item`. For different items you may get different contracts.